src/StoreApi/Schemas/AbstractAddressSchema.php

```diff
@@ -0,0 +1,181 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\StoreApi\Schemas;
+
+use Automattic\WooCommerce\Blocks\RestApi\Routes;
+
+/**
+ * AddressSchema class.
+ *
+ * Provides a generic address schema for composition in other schemas.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ * @since 4.1.0
+ */
+abstract class AbstractAddressSchema extends AbstractSchema {
```

Comment:
I don't want to block the pull request so this is mostly just food for thought.

I have seen validation and sanitization implemented via the [strategy design pattern](https://refactoring.guru/design-patterns/strategy/php/example) where you inject the Santization/Validation strategy used (that implements a defined interface). So for example there could be a `ValidationInterface` and `SanitizationInterface` and various implementations of those interfaces (of which `Address` would be one). `RequestAddressValidation` or `RequestAddressSanitization` in turn could be compositions of other things like `EmailValidation` etc.

Then, your schemas would just receive the validation and sanitization strategies to use injected into the constructors.

This could feasibly allow for the validation and sanitization strategies to be re-used in multiple places needing validation or sanitization.

I realize implementing something like this right now might be overkill or premature, but thought I'd mention anyways in case you come across places where this could potentially be implemented in the future.

Reply:
I think shared validation would be a nice idea for the logic now within OrderController! 

At schema level the validation and sanitised is directly coupled to the defined schema, meaning if schema changes, sanitisation and validation also needs changing, so I think there is less need for reuse. To clarify; the validation callbacks within the Schema classes validate against the defined Schema. e.g. Schema defines an enum, and the validation validates against that same enum.